src/coreclr/jit/emit.cpp

```diff
@@ -1206,7 +1206,7 @@ float emitter::insEvaluateExecutionCost(instrDesc* id)
     assert(throughput >= 0.0);
     assert(latency >= 0.0);
 
-    if (memAccessKind == PERFSCORE_MEMORY_WRITE)
+    if (memAccessKind == PERFSCORE_MEMORY_WRITE || memAccessKind == PERFSCORE_MEMORY_READ_WRITE)
```

Comment:
We're only hiding the write part of R/W here so that `add [mem], x` doesn't end up with a higher perfscore than `mov reg, [mem]; add reg, x; mov [mem], reg`.
The assumption below is now assumed to be true for both writes and I've adjusted some of the instruction latencies to reflect that.